docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -346,58 +387,72 @@ spec:
+##### KubeadmControlPlane rollout
 
-- Rollout operations rely on scale up and scale down which are be blocked based on Etcd and control plane health checks
+KubeadmControlPlane rollout operations rely on [scale up](#scale up) and [scale down](#scale_down) which are be blocked based on Etcd and control plane health checks
   - See [Health checks](#Health checks) below.
 
-- The rollout algorithm is the following:
-  - Find Machines that have an outdated spec
-  - If there is a machine requiring rollout
-    - Scale up control plane creating a machine with the new spec
-    - Scale down control plane by removing one of the machine that needs rollout (the oldest out-of date machine in the failure domain that has the most control-plane machines on it)
-    
+KubeadmControlPlane rollout is triggered by:
+  - Changes to Version
+  - Changes to the kubeadmConfigSpec
+  - Changes to the infrastructureRef
+  - The `upgradeAfter` field, which can be set to a specific time in the future
+    - Set to `nil` or the zero value of `time.Time` if no upgrades are desired
+    - An upgrade will run when that timestamp is passed
+    - Good for scheduling upgrades/SLOs
+    - Set `upgradeAfter` to now (in RFC3339 form) if an upgrade is required immediately
+
+- The controller should tolerate the manual or automatic removal of a replica during the upgrade process. A replica that fails during the upgrade may block the completion of the upgrade. Removal or other remedial action may be necessary to allow the upgrade to complete.
+
 - In order to determine if a Machine to be rolled out, KCP implements the following:
     - The infrastructureRef link used by each machine at creation time is stored in annotations at machine level.
     - The kubeadmConfigSpec used by each machine at creation time is stored in annotations at machine level.
-        - If the annotation is not present (machine is either old or adopted), we won't roll out on any possible changes made in KCP's ClusterConfiguration given that we don't have enough information to make a decision.
-           Users should use KCP.Spec.UpgradeAfter field to force a rollout in this case.
-    
-- The controller should tolerate the manual or automatic removal of a replica during the upgrade process. A replica that fails during the upgrade may block the completion of the upgrade. Removal or other remedial action may be necessary to allow the upgrade to complete.
+        - If the annotation is not present (machine is either old or adopted), we won't roll out on any possible changes made in KCP's ClusterConfiguration given that we don't have enough information to make a decision. Users should use KCP.Spec.UpgradeAfter field to force a rollout in this case.
+
+##### RolloutType ScaleOut (create-swap-and-delete)
+
+Create-swap-and-delete rollout strategy is performed when `KubeadmControlPlane.Spec.RolloutStrategy` is not set or `KubeadmControlPlane.Spec.RolloutStrategy` is set to "ScaleOut".  
```

Comment:
Instead of relying of "unset" behavior, the value should be defaulted if it's not present

Reply:
What's the `swap` in `Create-swap-and-delete` referring to?

Reply:
Here, the strategy is `ScaleOut`, which is not one of the available options outlined above